src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 200) {
+    this.health = health;
+    this.name = 'Animal';
+  }
+
+  dead(healthPoints) {
+    this.health += healthPoints;
+
+    if (this.health < 100) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health);
```

Comment:
[CODE PERFORMANCE] The filter operation inside the `dead` method is inefficient because it's inside a loop that's called every time an animal's health is updated. Consider moving the logic that removes dead animals from the `Animal.alive` array outside of this method to avoid unnecessary looping.